src/actor/component.rs

```diff
@@ -52,9 +55,18 @@ impl host::Host for Ctx<'_> {
         operation: String,
         payload: Option<Vec<u8>>,
     ) -> anyhow::Result<Result<Option<Vec<u8>>, String>> {
-        let invocation = (namespace, operation, payload)
+        let invocation: Invocation = (namespace, operation, payload)
             .try_into()
             .context("failed to parse invocation")?;
+        //TODO: ugly hacky try make better, maybe better .into above
+        let invocation: Invocation = if let Invocation::Host(inv) = invocation {
+            Invocation::Host(HostInvocation {
+                call_context: self.call_context.clone(),
+                ..inv
+            })
+        } else {
+            invocation
+        };
```

Comment:
None of this should be necessary, instead, the context should be passed as additional parameter to `handle`, since that's a property of an instance of an actor and not the individual invocation

Reply:
Done!